test-crawler/collector/collector_test.go

```diff
@@ -0,0 +1,42 @@
+package collector
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListGoFilesInFolder(t *testing.T) {
+	t.Run("Read golang files from temporary directory", func(t *testing.T) {
+		tempDir := t.TempDir()
+		err := ioutil.WriteFile(fmt.Sprintf("%s/a.go", tempDir), []byte("package main func main(){a:=1}"), 0644)
+		assert.NoError(t, err)
+		err = ioutil.WriteFile(fmt.Sprintf("%s/b.go", tempDir), []byte("package main func main(){b:=2}"), 0644)
+		assert.NoError(t, err)
+
+		system, files, err := ListGoFilesInFolder(tempDir, nil)
+		assert.NoError(t, err)
+		splitedFolderName := strings.Split(tempDir, "/")
+		systemName := splitedFolderName[len(splitedFolderName)-1]
+		assert.Equal(t, system, systemName)
+		assert.Equal(t, 2, len(files))
+	})
+
+	t.Run("Read golang files from mocks/config directory", func(t *testing.T) {
+		system, files, err := ListGoFilesInFolder("../mocks/config", nil)
+		assert.NoError(t, err)
+		assert.Equal(t, system, "config")
+		assert.Equal(t, 2, len(files))
+	})
+
+	t.Run("Read golang files from mocks directory", func(t *testing.T) {
+		system, files, err := ListGoFilesInFolder("../mocks", nil)
+		assert.NoError(t, err)
+		assert.Equal(t, system, "mocks")
+		assert.Equal(t, 4, len(files))
```

Comment:
This test is fragile - suppose someone adds a new file to the `mocks` or `config` folder, to test a feature that is unrelated to this one. Adding those files would cause this test to fail, even though the code changes did not affect this feature.

Use the `t.TempDir` approach, and create new temp files.

Reply:
Another key point - the names of the tests tell us nothing about what they are covering. The tests should be named after the behavior they are actually testing.

Reply:
you are right, nice catch, thanks!